### PR TITLE
Emit aggregate row id for the MoveNext method

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/53758

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueStateMachineTests.cs [hunk 1]
```diff
@@ -2624,6 +2650,35 @@ static async Task<int> F()
...
+            <entry offset=""0x79"" startLine=""13"" startColumn=""9"" endLine=""13"" endColumn=""20"" document=""1"" />
+            <entry offset=""0x85"" hidden=""true"" document=""1"" />
+            <entry offset=""0xd8"" startLine=""14"" startColumn=""9"" endLine=""14"" endColumn=""18"" document=""1"" />
+            <entry offset=""0xdc"" hidden=""true"" document=""1"" />
+            <entry offset=""0xf6"" startLine=""15"" startColumn=""5"" endLine=""15"" endColumn=""6"" document=""1"" />
+            <entry offset=""0xfe"" hidden=""true"" document=""1"" />
+          </sequencePoints>
+          <asyncInfo>
+            <kickoffMethod token=""0x6000004"" />
+            <await yield=""0x37"" resume=""0x55"" token=""0x6000009"" />
```
Comment (davidwengier) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueStateMachineTests.cs:2675]: Before my change the token here was for row id 3

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueStateMachineTests.cs [hunk 2]
```diff
@@ -2779,6 +2834,35 @@ .maxstack  3
...
```
Comment (davidwengier) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueStateMachineTests.cs:2859]: Before the change this was for row id 2

**Comment by davidwengier** [comment]
Updated to test native PDBs too. Looks like the bug wasn't present there.

**Review by tmat — APPROVED** [review]
:shipit:

**Review by cston — APPROVED** [review]

**Comment by davidwengier** [comment]
@dotnet/roslyn-compiler for a second review

**Review by RikkiGibson — APPROVED** [review]

**Pull request merged**